nntrainer/models/neuralnet.cpp

```diff
@@ -162,31 +164,71 @@ int NeuralNetwork::initialize() {
  */
 NeuralNetwork::~NeuralNetwork() { model_graph.reset(); }
 
-void NeuralNetwork::setLabels(sharedConstTensors label) {
-  auto fill_label = [&label](auto const &layer_node) {
-    NNTR_THROW_IF(label.size() != layer_node->getNumOutputs(),
-                  std::invalid_argument)
-      << "label size does not match with the layer requirements"
-      << " layer: " << layer_node->getName() << " label size: " << label.size()
-      << " requirements size: " << layer_node->getNumOutputs();
+static void setLabels(const std::vector<Tensor> &data,
+                      const std::vector<Var_Grad *> &label_list) {
 
-    for (unsigned int i = 0; i < layer_node->getNumOutputs(); i++) {
-      layer_node->getOutputGradUnsafe(i) = *label[i];
-    }
-  };
+  NNTR_THROW_IF(!data.empty() && data.size() != label_list.size(),
```

Comment:
Data.size() == 1 is not an error?

Reply:
This function will be slightly refactored when I borrow the latest version from #1544. Please refer to https://github.com/nnstreamer/nntrainer/pull/1544/files#diff-8136da3afc181f3163a8f8a15d86d132abee8c07de2c9b49bc2cf0759e380a69R833-R879 for the reference code. I will update as per this soon in this PR.

Reply:
Cool, no worries. You can delay to reduce work at will :)

Reply:
great, then I will let #1544 resolve it.